docs/writing_to_promscale.md

```diff
@@ -0,0 +1,173 @@
+
+
+## General time-series format rules
+A time-series must always contain both labels and samples.
+
+
+Labels are a collection of label name/value pairs with unique label names. Every request must  contain a label named `__name__` which cannot have an empty value. All other labels are optional metadata.
+
+
+A series entry always contains two things:
+* An integer timestamp in milliseconds since epoch, i.e. 1970-01-01 00:00:00 UTC, excluding leap second, represented as required by Go's [ParseInt](https://golang.org/pkg/strconv/#ParseInt) function. 
+* Floating point number that represents the actual measured value.
+
+## JSON streaming format
+
+This format was introduced in Promscale to enable easier usage of the endpoint when ingesting metric data from 3rd party tools. It is not part of the `remote_write` specification for Prometheus. It is slightly less efficient to use this format than the Protobuf format. 
+
+
+This format is a JSON stream of objects with two fields:
+* `labels` JSON object in which all the labels are represented as fields
+* `samples` JSON array which contains arrays of timestamp-value sets
+
+Here is a short example of what a simple payload would look like:
+
+```
+{
+    "labels":{"__name__": "cpu_usage", "namespace":"dev", "node": "brain"},
+    "samples":[
+        [1577836800000, 100],
+        [1577836801000, 99],
+        [1577836802000, 98],
+        ...
+    ]
+}
+{
+    "labels":{"__name__": "cpu_usage", "namespace":"prod", "node": "brain"},
+    "samples":[
+        [1577836800000, 98],
+        [1577836801000, 99],
+        [1577836802000, 100],
+        ...
+    ]
+}
+...
+```
+
+As you can see, the labels object is pretty self explanatory. Fields represent the label name and value is the label value. Samples array consists of arrays with some strict rules:
+* There can be only two values in the array
+* First value is a UNIX timestamp in milliseconds which has to be an integer value
+* Second value is a floating point number which represents the corresponding value of the metric.
+
+
+In order to send a request to Promscale, you would need to send an HTTP POST request with the request body set to the JSON payload and set the required header values:
+* `Content-Type` header should be set to `application/json`
+* `Content-Encoding` header can optionally be set to `snappy` if you are using snappy compression on the JSON payload
+
+
+Here is an example of a request sent using `curl` tool:
+
+```
+curl --header "Content-Type: application/json" \
+--request POST \
+--data '{"labels":{"__name__":"foo"},"samples":[[1577836800000, 100]]}' \
+http://localhost:3000/api/login
+```
+
+## Protobuf format
+
+This format is a little more complex and involved to use than the JSON format but generally is a bit more performant because the payload is smaller in size. It is recommended that you use this format if you are building a system which would ingest a lot of data continuously.
```

Comment:
To be strictly correct, should we say *more performant* or *more efficient*? Because performance in proto-bufs can decreases due to the involved Unmarshalling/decoding mechanism. That means, it's only efficient (in space) and not performant.

Reply:
I don't think performant has a space and time definition. In general, I think performant and efficient are complete synonyms so it doesn't matter. We also explain `because the payload is smaller in size` explicitly